streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -548,24 +548,13 @@ private KStream<K, V> doRepartition(final Repartitioned<K, V> repartitioned) {
 
     @Override
     public void to(final String topic) {
-        to(topic, Produced.with(keySerde, valueSerde, null));
+        to(new StaticTopicNameExtractor<>(topic), Produced.with(keySerde, valueSerde, null));
     }
 
-    @SuppressWarnings("resource")
     @Override
     public void to(final String topic,
                    final Produced<K, V> produced) {
-        Objects.requireNonNull(topic, "topic can't be null");
-        Objects.requireNonNull(produced, "produced can't be null");
```

Comment:
Unifying this check, plus `ProducedInternal` code below into internal helper method that is called by all public methods.